Sources/Public/DotLottie/DotLottieFileHelpers.swift

```diff
@@ -9,26 +9,40 @@ import Foundation
 
 extension DotLottieFile {
 
-  // MARK: Public
-
-  /// A closure for an DotLottieFile download. The closure is passed `nil` if there was an error.
-  public typealias DotLottieLoadClosure = (Result<DotLottieFile, Error>) -> Void
-
-  // MARK: DotLottie file (Loading)
+  /// Loads a DotLottie model from a bundle by its name. Returns `nil` if a file is not found.
+  ///
+  /// - Parameter name: The name of the lottie file without the lottie extension. EG "StarAnimation"
+  /// - Parameter bundle: The bundle in which the lottie is located. Defaults to `Bundle.main`
+  /// - Parameter subdirectory: A subdirectory in the bundle in which the lottie is located. Optional.
+  /// - Parameter dotLottieCache: A cache for holding loaded lotties. Defaults to `LRUDotLottieCache.sharedCache`. Optional.
+  @available(iOS 13.0, macOS 10.15, tvOS 13.0, *)
+  public static func named(
+    _ name: String,
+    bundle: Bundle = Bundle.main,
+    subdirectory: String? = nil,
+    dotLottieCache: DotLottieCacheProvider? = DotLottieCache.sharedCache) async throws
+    -> DotLottieFile
+  {
+    try await withCheckedThrowingContinuation { continuation in
+      DotLottieFile.named(name, bundle: bundle, subdirectory: subdirectory, dotLottieCache: dotLottieCache) { result in
+        continuation.resume(with: result)
+      }
+    }
+  }
 
   /// Loads a DotLottie model from a bundle by its name. Returns `nil` if a file is not found.
   ///
   /// - Parameter name: The name of the lottie file without the lottie extension. EG "StarAnimation"
   /// - Parameter bundle: The bundle in which the lottie is located. Defaults to `Bundle.main`
   /// - Parameter subdirectory: A subdirectory in the bundle in which the lottie is located. Optional.
-  /// - Parameter closure: A closure to be called when the file has loaded.
   /// - Parameter dotLottieCache: A cache for holding loaded lotties. Defaults to `LRUDotLottieCache.sharedCache`. Optional.
+  /// - Parameter handleResult: A closure to be called when the file has loaded.
   public static func named(
     _ name: String,
     bundle: Bundle = Bundle.main,
     subdirectory: String? = nil,
-    closure: @escaping DotLottieLoadClosure,
-    dotLottieCache: DotLottieCacheProvider? = DotLottieCache.sharedCache)
+    dotLottieCache: DotLottieCacheProvider? = DotLottieCache.sharedCache,
+    handleResult: @escaping (Result<DotLottieFile, Error>) -> Void)
   {
     DispatchQueue.global().async {
```

Comment:
Seems like if you wanted you could potentially check for task cancellation in these methods via `Task.isCancelled`  to support async cancellation